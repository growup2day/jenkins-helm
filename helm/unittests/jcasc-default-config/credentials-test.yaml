templates:
  - config-jcasc.yaml
tests:
  - it: should include credentials config in default config scripts when configured
    set:
      JCasC:
        defaultConfig: true
        vault:
          vaultUrl: "https://vault.nzlb.service.anz:8200"
          credentialId: vault-auth
          namespace: nonprod
          mountPath: kubernetes
          role: microservice
          credentials:
            artifactory-api-key:
              kind: vaultUsernamePasswordCredentialImpl
              scope: GLOBAL
              description: "API Key for Artifactory"
              path:
                'secret/context/{{ .Release.Namespace }}/{{ include "jenkins.serviceAccountName" .
                }}/pairs/artifactory-api-key'
            fortify-ci-token:
              kind: vaultStringCredentialImpl
              scope: GLOBAL
              description: "CI Token for Fortify"
              path: 'secret/context/{{ .Release.Namespace }}/{{ include "jenkins.serviceAccountName" . }}/keys'
              vaultKey: "fortify-ci-token"

    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-credentials.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-credentials.yaml]
  - it: should include just the Vault credentials config in default config scripts when not configured
    set:
      JCasC:
        defaultConfig: true
    asserts:
      - isNotNull:
          path: data.[jcasc-default-config-credentials.yaml]
      - matchSnapshot:
          path: data.[jcasc-default-config-credentials.yaml]
  - it: should include empty list of credentials in default config scripts when Vault is disabled
    set:
      JCasC:
        defaultConfig: true
        vault:
          enabled: false
    asserts:
      - matchSnapshot:
          path: data.[jcasc-default-config-credentials.yaml]
  - it:
      should not include credentials config in default config scripts when there is custom config script for the section
    set:
      JCasC:
        defaultConfig: true
        configScripts:
          custom-override: |
            credentials:
              system:
                domainCredentials:
                  - credentials: []
    asserts:
      - isNull:
          path: data.[jcasc-default-config-credentials.yaml]
      - isNotNull:
          path: data.[custom-override.yaml]
