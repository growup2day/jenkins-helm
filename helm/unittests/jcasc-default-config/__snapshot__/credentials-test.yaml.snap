should include credentials config in default config scripts when configured:
  1: |
    |-
      # Manage Credentials
      credentials:
        system:
          domainCredentials:
            - credentials:
                - vaultKubernetesCredential:
                    description: "Vault Auth"
                    id: "vault-auth"
                    mountPath: "kubernetes"
                    role: "microservice"
                    scope: "SYSTEM"
                - vaultUsernamePasswordCredentialImpl:
                    id: "artifactory-api-key"
                    description: API Key for Artifactory
                    path: secret/context/NAMESPACE/RELEASE-NAME-jenkins/pairs/artifactory-api-key
                    scope: GLOBAL
                - vaultStringCredentialImpl:
                    id: "fortify-ci-token"
                    description: CI Token for Fortify
                    path: secret/context/NAMESPACE/RELEASE-NAME-jenkins/keys
                    scope: GLOBAL
                    vaultKey: fortify-ci-token
should include empty list of credentials in default config scripts when Vault is disabled:
  1: |
    |-
      # Manage Credentials
      credentials:
        system:
          domainCredentials:
            - credentials: []
should include just the Vault credentials config in default config scripts when not configured:
  1: |
    |-
      # Manage Credentials
      credentials:
        system:
          domainCredentials:
            - credentials:
                - vaultKubernetesCredential:
                    description: "Vault Auth"
                    id: "vault-auth"
                    mountPath: "kubernetes"
                    role: "microservice"
                    scope: "SYSTEM"
